examples/fluids/problems/vortexshedding.c

```diff
@@ -0,0 +1,148 @@
+    //problem->apply_freestream.qfunction_loc = Freestream_Conserv_loc;
+  case STATEVAR_PRIMITIVE:
+    problem->ics.qfunction      = ICsVortexshedding_Prim;
+    problem->ics.qfunction_loc  = ICsVortexshedding_Prim_loc;
+    //problem->apply_freestream.qfunction     = Freestream_Prim;
+    //problem->apply_freestream.qfunction_loc = Freestream_Prim_loc;
+  }
+
+  CeedScalar U_infty[3]         = {1.0};       // m/s
+  CeedScalar T_infty            = 300.;        // K
+  CeedScalar P0                 = 1.e5;        // Pa
+  CeedScalar L;
+  CeedScalar H;
+  CeedScalar D;
+
+  PetscOptionsBegin(comm, NULL, "Options for VORTEX SHEDDING problem", NULL);
+  PetscInt narray=3;
+  ierr = PetscOptionsScalarArray("-freestream_velocity", "Velocity at freestream",
+                            NULL, U_infty, &narray, NULL); CHKERRQ(ierr);
+  PetscCheck(narray == 3, comm, PETSC_ERR_ARG_SIZ,
+             "-freestream_velocity should recieve array of size 3, instead recieved size %"
+             PetscInt_FMT".", narray);
+  ierr = PetscOptionsScalar("-freestream_pressure", "Pressure at freestream",
+                            NULL, P0, &P0, NULL); CHKERRQ(ierr);
+  ierr = PetscOptionsScalar("-freestream_temperature", "Temperature at freestream",
+                            NULL, T_infty, &T_infty, NULL); CHKERRQ(ierr);
+  ierr = PetscOptionsScalar("-L", "Length of the rectangular channel",
+                            NULL, L, &L, NULL); CHKERRQ(ierr);
+  ierr = PetscOptionsScalar("-H", "Heigth of the rectangular channel",
+                            NULL, H, &H, NULL); CHKERRQ(ierr);
+  ierr = PetscOptionsScalar("-D", "Cylinder diameter",
+                            NULL, D, &D, NULL); CHKERRQ(ierr);
+  PetscOptionsEnd();
+
+  PetscScalar meter  = user->units->meter;
+  PetscScalar second = user->units->second;
+  PetscScalar Kelvin = user->units->Kelvin;
+  PetscScalar Pascal = user->units->Pascal;
+
+  T_infty *= Kelvin;
+  P0      *= Pascal;
+  for (int i=0; i<3; i++) {
+    U_infty[i]     *= meter / second;
+  }
+  L      *= meter;
+  H      *= meter;
+  D      *= meter;
+
+  CeedQFunctionContextGetData(problem->apply_vol_rhs.qfunction_context,
+                              CEED_MEM_HOST, &newtonian_ig_ctx);
+  State S_infty;
+  {
+    CeedScalar Y[5] = {P0, U_infty[0], U_infty[1], U_infty[2], T_infty};
+    CeedScalar x[3] = {0.};
+    S_infty = StateFromY(newtonian_ig_ctx, Y, x);
+  }
+
+  //-- Setup Problem information
+  CeedScalar center;
+  {
+    PetscReal domain_min[3], domain_max[3], domain_size[3];
+    ierr = DMGetBoundingBox(dm, domain_min, domain_max); CHKERRQ(ierr);
+    // compute L and H
+    for (PetscInt i=0; i<3; i++) domain_size[i] = domain_max[i] - domain_min[i];
+    L = domain_size[0]*meter;
+    H = domain_size[1]*meter;
+    center = 0.5 * H; // this should be checked
+  }
+  // Some properties depend on parameters from NewtonianIdealGas
+  CeedQFunctionContextGetData(problem->apply_vol_rhs.qfunction_context,
```

Comment:
You're getting an error saying you're trying to access this context while its locked, because you already got access (that you haven't restored yet) a few lines above.

Delete the second occurrence to fix this issue.